Index: arch/arm/boot/dts/Makefile
===================================================================
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -922,14 +922,22 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32746g-eval.dtb \
 	stm32h743i-eval.dtb \
 	stm32h743i-disco.dtb \
+	stm32mp153a-txmp-1530.dtb \
+	stm32mp153a-txmp-1530-mb7.dtb \
+	stm32mp153a-txmp-1530-mipi-mb.dtb \
 	stm32mp157a-dk1.dtb \
+	stm32mp157c-qsmp-1570.dtb \
+	stm32mp157c-qsmp-1570-qsbase1.dtb \
 	stm32mp157c-dk2.dtb \
 	stm32mp157c-dk2-a7-examples.dtb \
 	stm32mp157c-dk2-m4-examples.dtb \
 	stm32mp157c-ed1.dtb \
 	stm32mp157c-ev1.dtb \
 	stm32mp157c-ev1-a7-examples.dtb \
-	stm32mp157c-ev1-m4-examples.dtb
+	stm32mp157c-ev1-m4-examples.dtb \
+	stm32mp157c-txmp-1570.dtb \
+	stm32mp157c-txmp-1570-mb7.dtb \
+	stm32mp157c-txmp-1570-mipi-mb.dtb
 dtb-$(CONFIG_MACH_SUN4I) += \
 	sun4i-a10-a1000.dtb \
 	sun4i-a10-ba10-tvbox.dtb \
