Index: karo-tx-linux/arch/arm/boot/dts/stm32mp157c.dtsi
===================================================================
--- karo-tx-linux.orig/arch/arm/boot/dts/stm32mp157c.dtsi
+++ karo-tx-linux/arch/arm/boot/dts/stm32mp157c.dtsi
@@ -1681,6 +1681,8 @@
 		};
 
 		qspi: qspi@58003000 {
+			#address-cells = <1>;
+			#size-cells = <0>;
 			compatible = "st,stm32f469-qspi";
 			reg = <0x58003000 0x1000>, <0x70000000 0x10000000>;
 			reg-names = "qspi", "qspi_mm";
@@ -1730,12 +1732,6 @@
 			status = "disabled";
 		};
 
-		stmmac_axi_config_0: stmmac-axi-config {
-			snps,wr_osr_lmt = <0x7>;
-			snps,rd_osr_lmt = <0x7>;
-			snps,blen = <0 0 0 0 16 8 4>;
-		};
-
 		ethernet0: ethernet@5800a000 {
 			compatible = "st,stm32mp1-dwmac", "snps,dwmac-4.20a";
 			reg = <0x5800a000 0x2000>;
@@ -1954,6 +1950,12 @@
 		};
 	};
 
+	stmmac_axi_config_0: stmmac-axi-config {
+		snps,wr_osr_lmt = <0x7>;
+		snps,rd_osr_lmt = <0x7>;
+		snps,blen = <0 0 0 0 16 8 4>;
+	};
+
 	m4_rproc: m4@0 {
 		compatible = "st,stm32mp1-rproc";
 		#address-cells = <1>;
Index: karo-tx-linux/arch/arm/boot/dts/stm32mp157-pinctrl.dtsi
===================================================================
--- karo-tx-linux.orig/arch/arm/boot/dts/stm32mp157-pinctrl.dtsi
+++ karo-tx-linux/arch/arm/boot/dts/stm32mp157-pinctrl.dtsi
@@ -834,7 +834,7 @@
 				};
 			};
 
-			m_can1_sleep_pins_a: m_can1-sleep@0 {
+			m_can1_sleep_pins_a: m_can1-sleep-0 {
 				pins {
 					pinmux = <STM32_PINMUX('H', 13, ANALOG)>, /* CAN1_TX */
 						 <STM32_PINMUX('I', 9, ANALOG)>; /* CAN1_RX */
@@ -1038,7 +1038,7 @@
 				};
 			};
 
-			rtc_out2_rmp_pins_a: rtc-out2-rmp-pins@0 {
+			rtc_out2_rmp_pins_a: rtc-out2-rmp-pins-0 {
 				pins {
 					pinmux = <STM32_PINMUX('I', 8, ANALOG)>; /* RTC_OUT2_RMP */
 				};
Index: karo-tx-linux/drivers/gpu/drm/stm/ltdc.c
===================================================================
--- karo-tx-linux.orig/drivers/gpu/drm/stm/ltdc.c
+++ karo-tx-linux/drivers/gpu/drm/stm/ltdc.c
@@ -587,7 +587,7 @@ static void ltdc_crtc_mode_set_nofb(stru
 	if (vm.flags & DISPLAY_FLAGS_DE_LOW)
 		val |= GCR_DEPOL;
 
-	if (vm.flags & DISPLAY_FLAGS_PIXDATA_POSEDGE)
+	if (vm.flags & DISPLAY_FLAGS_PIXDATA_NEGEDGE)
 		val |= GCR_PCPOL;
 
 	reg_update_bits(ldev->regs, LTDC_GCR,
