Index: u-boot/common/fdt_support.c
===================================================================
--- u-boot.orig/common/fdt_support.c
+++ u-boot/common/fdt_support.c
@@ -636,6 +636,8 @@ int fdt_shrink_to_minimum(void *blob, ui
 	ret = fdt_add_mem_rsv(blob, (uintptr_t)blob, actualsize);
 	if (ret < 0)
 		return ret;
+	if (env_get("fdtsize"))
+		env_set_hex("fdtsize", actualsize);
 
 	return actualsize;
 }
